Modules/_zoneinfo.c

```diff
@@ -2603,14 +2604,16 @@ module_free()
 
     xdecref_ttinfo(&NO_TTINFO);
 
-    Py_XDECREF(TIMEDELTA_CACHE);
-    if (!Py_REFCNT(TIMEDELTA_CACHE)) {
-        TIMEDELTA_CACHE = NULL;
+    if (TIMEDELTA_CACHE != NULL && Py_REFCNT(TIMEDELTA_CACHE) > 1) {
+        Py_DECREF(TIMEDELTA_CACHE);
+    } else {
```

Comment:
Ugh, I just realize this is a PEP 7 violation, it should be:

```c
    if (TIMEDELTA_CACHE != NULL && Py_REFCNT(TIMEDELTA_CACHE) > 1) {
        Py_DECREF(TIMEDELTA_CACHE);
    }
    else {
        Py_CLEAR(TIMEDELTA_CACHE);
    }
```

Since we're probably going to re-write this code soon-ish anyway, I guess we'll just leave it here and in the backport until someone touches it again.

Reply:
Aah sorry about that :(